llama_index/indices/response/response_builder.py

```diff
@@ -576,6 +587,122 @@ def get_response(
             return stream_response
 
 
+class Accumulate(BaseResponseBuilder):
+    def __init__(
+        self,
+        service_context: ServiceContext,
+        text_qa_template: QuestionAnswerPrompt,
+        streaming: bool = False,
+        use_async: bool = False,
+    ) -> None:
+        super().__init__(service_context=service_context, streaming=streaming)
+        self.text_qa_template = text_qa_template
+        self._use_async = use_async
+
+    def flatten_list(self, md_array: List[List[Any]]) -> List[Any]:
+        return list(item for sublist in md_array for item in sublist)
+
+    def format_response(self, outputs: List[Any], separator: str) -> str:
+        responses: List[str] = []
+        for response, formatted_prompt in outputs:
+            self._log_prompt_and_response(
+                formatted_prompt, response, log_prefix="Initial"
+            )
+            responses.append(response or "Empty Response")
+
+        return separator.join(
+            [f"Response {index + 1}: {item}" for index, item in enumerate(responses)]
+        )
+
+    @llm_token_counter("aget_response")
+    async def aget_response(
+        self,
+        query_str: str,
+        text_chunks: Sequence[str],
+        separator: str = "\n---------------------\n",
+        **kwargs: Any,
+    ) -> RESPONSE_TEXT_TYPE:
+        """Apply the same prompt to text chunks and return async responses"""
+
+        tasks = self.get_response(query_str, text_chunks, **kwargs)
+
+        flattened_tasks = self.flatten_list(tasks)
+        outputs = await asyncio.gather(*flattened_tasks)
+
+        return self.format_response(outputs, separator)
+
+    @llm_token_counter("get_response")
+    def get_response(
+        self,
+        query_str: str,
+        text_chunks: Sequence[str],
+        separator: str = "\n---------------------\n",
+    ) -> str | List[List[Any]]:
```

Comment:
I think lint will complain about this, since it's not available in earlier python versions.